src/Altinn.Platform/Altinn.Platform.Storage/Storage/Repository/InstanceRepository.cs

```diff
@@ -88,51 +97,293 @@ public async Task<bool> Delete(Instance item)
+            if (queryValue.StartsWith("gt:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.DueDateTime > dateValue);
+            }
 
-            return instances;
+            if (queryValue.StartsWith("gte:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(4));
+                return queryBuilder.Where(i => i.DueDateTime >= dateValue);
+            }
+
+            if (queryValue.StartsWith("lt:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.DueDateTime < dateValue);
+            }
+
+            if (queryValue.StartsWith("lte:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(4));
+                return queryBuilder.Where(i => i.DueDateTime <= dateValue);
+            }
+
+            if (queryValue.StartsWith("eq:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.DueDateTime == dateValue);
+            }
+
+            dateValue = ParseDateTimeIntoUtc(queryValue);
+            return queryBuilder.Where(i => i.DueDateTime == dateValue); 
+        }
+
+        private IQueryable<Instance> QueryBuilderForLastChangedDateTime(IQueryable<Instance> queryBuilder, string queryValue)
+        {
+            DateTime dateValue;
+
+            if (queryValue.StartsWith("gt:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.LastChangedDateTime > dateValue);
+            }
+
+            if (queryValue.StartsWith("gte:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(4));
+                return queryBuilder.Where(i => i.LastChangedDateTime >= dateValue);
+            }
+
+            if (queryValue.StartsWith("lt:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.LastChangedDateTime < dateValue);
+            }
+
+            if (queryValue.StartsWith("lte:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(4));
+                return queryBuilder.Where(i => i.LastChangedDateTime <= dateValue);
+            }
+
+            if (queryValue.StartsWith("eq:"))
+            {
+                dateValue = ParseDateTimeIntoUtc(queryValue.Substring(3));
+                return queryBuilder.Where(i => i.LastChangedDateTime == dateValue);
+            }
+
+            dateValue = ParseDateTimeIntoUtc(queryValue);
+            return queryBuilder.Where(i => i.LastChangedDateTime == dateValue);
+        }
+
+        private IQueryable<Instance> QueryBuilderForCreatedDateTime(IQueryable<Instance> queryBuilder, string queryValue)
```

Comment:
a lot of duplicated code when parsing the date time, could we add a parameter to identify if it is visibleDateTime or createdDateTime and use a single function to generate the query?

Reply:
I agree, but could not find a way to avoid it. Cosmos QueryBuilder does not support Invoke methods in where clauses.

Reply:
Have made comment in code